Keen/AccessKey/AccessKeys.cs

```diff
@@ -0,0 +1,104 @@
+    /// </summary>
+    public class AccessKeys : IAccessKeys
+    {
+        private readonly IKeenHttpClient _keenHttpClient;
+        private readonly string _acceskeyRelativeUrl;
+        private readonly string _readKey;
+        private readonly string _masterKey;
+
+        internal AccessKeys(IProjectSettings prjSettings,
+                       IKeenHttpClientProvider keenHttpClientProvider)
+        {
+            if (null == prjSettings)
+            {
+                throw new ArgumentNullException(nameof(prjSettings),
+                                                "Project Settings must be provided.");
+            }
+
+            if (null == keenHttpClientProvider)
+            {
+                throw new ArgumentNullException(nameof(keenHttpClientProvider),
+                                                "A KeenHttpClient provider must be provided.");
+            }
+
+            if (string.IsNullOrWhiteSpace(prjSettings.KeenUrl) ||
+                !Uri.IsWellFormedUriString(prjSettings.KeenUrl, UriKind.Absolute))
+            {
+                throw new KeenException(
+                    "A properly formatted KeenUrl must be provided via Project Settings.");
+            }
+
+            var serverBaseUrl = new Uri(prjSettings.KeenUrl);
+            _keenHttpClient = keenHttpClientProvider.GetForUrl(serverBaseUrl);
+            _acceskeyRelativeUrl = KeenHttpClient.GetRelativeUrl(prjSettings.ProjectId,
+                                                               KeenConstants.AccessKeyResource);
+
+            _readKey = prjSettings.ReadKey;
+            _masterKey = prjSettings.MasterKey;
+        }
+
+        public async Task<JObject> CreateAccessKey(AccessKey accesskey)
+        {
+            if (string.IsNullOrWhiteSpace(_masterKey))
+            {
+                throw new KeenException("An API WriteKey is required to add events.");
+            }
+
+            var xcontent = accesskey.ToString();
+
+            DefaultContractResolver contractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new SnakeCaseNamingStrategy()
+            };
+
+            var content = JsonConvert.SerializeObject(accesskey, new JsonSerializerSettings
+            {
+                ContractResolver = contractResolver,
+                Formatting = Formatting.Indented
+            }).ToSafeString();
+
+            var responseMsg = await _keenHttpClient
+                .PostAsync(_acceskeyRelativeUrl, _masterKey, content)
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            var responseString = await responseMsg
+                .Content
+                .ReadAsStringAsync()
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            JObject jsonResponse = null;
+
+            try
+            {
+                jsonResponse = JObject.Parse(responseString);
+            }
+            catch (Exception)
```

Comment:
Was this because you were seeing a 404 error or something similar and it's actually OK in this case to swallow the exception? Then will we throw below? I know we do this elsewhere, and we should really keep the inner exception around for context, but that's another issue. At least if that's the case, let's put a comment here explaining the empty `catch(){}` block.